flink-ml-core/src/test/java/org/apache/flink/ml/common/datastream/TableUtilsTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.datastream;
+
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.Schema;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.types.Row;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.sql.Timestamp;
+
+/** Tests the {@link TableUtils}. */
+public class TableUtilsTest {
+    private StreamExecutionEnvironment env;
+    private StreamTableEnvironment tEnv;
+
+    @Before
+    public void before() {
+        env = StreamExecutionEnvironment.getExecutionEnvironment();
+        tEnv = StreamTableEnvironment.create(env);
+    }
+
+    @Test
+    public void testGetRowTypeInfo() {
+        Table inputTable =
+                tEnv.fromDataStream(
+                        env.fromElements(new Timestamp(0)),
```

Comment:
Let's also add tests to verify the conversion of other data types as well. So far as I can see, we should include tests for data types listed [here](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/types/#list-of-data-types) and the vector/matrix data types introduced by Flink ML.

Reply:
Thanks for the comment. I have included all predefined data types as well as vector/matrix in the test case.

One exception is that `DataTypes.TIMESTAMP_WITH_TIME_ZONE()` does not work in either `TypeInformation.of()` or `ExternalTypeInfo.of()`. It is not covered in the test case.